server/src/main/java/org/elasticsearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -230,10 +230,15 @@ public static void ensureIndexCompatibility(final Version nodeVersion, Metadata
         // we ensure that all indices in the cluster we join are compatible with us no matter if they are
         // closed or not we can't read mappings of these indices so we need to reject the join...
         for (IndexMetadata idxMetadata : metadata) {
-            if (idxMetadata.getCreationVersion().after(nodeVersion)) {
+            if (idxMetadata.getCreationVersion().afterMajorMinor(nodeVersion)) {
                 throw new IllegalStateException("index " + idxMetadata.getIndex() + " version not supported: "
                     + idxMetadata.getCreationVersion() + " the node version is: " + nodeVersion);
             }
+            if (nodeVersion.luceneVersion.onOrAfter(idxMetadata.getCreationVersion().luceneVersion) == false) {
```

Comment:
As no existing version pair would match this condition I see no way in testing this.

Reply:
We could create arbitrary versions if we exposed the `Version` constructor and using `org.apache.lucene.util.Version.fromBits` (or maybe there are fitting constants around). Not sure if it's worth it.

Reply:
That's exactly what I tried, but the version will be checked on the index's metadata version where we only store the main version, not including any lucene version. That means, it will lookup the Version from the registered ones to get the instance, thus we'd have to register that dummy version..

Reply:
Isn't the metadata version settable here https://github.com/crate/crate/pull/10924/files#diff-741e6c663352f5f936adbb2643745f7928aeec6a4d473aeea243719ca512736fR72 ?

Reply:
Yes but it will end up in a Version string, not the Version object. So on reading it will lookup the string on the registry.

Reply:
ah, got it. Let's skip the test case then.